app/controllers/oai_controller.rb

```diff
@@ -40,6 +42,14 @@ def render_list_sets
     render xml: build_list_sets_response(Organization.providers)
   end
 
+  def render_identify
+    headers['Cache-Control'] = 'no-cache'
+    headers['Last-Modified'] = Time.current.httpdate
+    headers['X-Accel-Buffering'] = 'no'
+
+    render xml: build_identify_response(Upload.order(:created_at).first.created_at)
```

Comment:
not sure this is right, but my understanding isn't very deep...we couldn't have a `created_at` that comes _after_ an `updated_at` though, so it does seem like this should give the earliest known upload from a provider.

Reply:
I'm not really sure either. This gets at our data retention policy and I'm not sure we're clear on that yet. Since we're building the oai response from the normalized dump from the default stream (for the moment, anyway) I wonder if we should use that to get the earliestDateTimestamp.
```suggestion
    render xml: build_identify_response(Stream.default.flat_map { |stream| stream.normalized_dumps }.reject { |dump| dump.oai_xml.nil? }.sort_by(&:created_at).first.created_at)
```

Reply:
Makes sense to me. I'm gonna see if I can do something like this on the ActiveRecord side, but if I can't make a reasonable query I'll use this method.

Reply:
Updated with a query that I think is right; I think the join to `oai_xml_attachment` gets the `.reject()` "for free"